Parsing/parse_wQR.py

```diff
@@ -1,77 +1,250 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import json
+import logging
+import os
+import re
+from datetime import datetime, timedelta
+from typing import Optional
+
+from pydantic import BaseModel, Field
 from pyzbar.pyzbar import decode, ZBarSymbol
 import cv2
 import sys
 import numpy as np
 import time
+import click
+
+# initialize the logger
+logger = logging.getLogger(__name__)
+logging.getLogger().addHandler(logging.StreamHandler(sys.stdout))
+stm_error = logging.StreamHandler(sys.stderr)
+stm_error.setLevel(logging.ERROR)
+logging.getLogger().addHandler(stm_error)
+logger.debug(f"name={__name__}")
+
+
+# Define class for video time info
+class VideoTimeInfo(BaseModel):
+    success: bool = Field(..., description="Success flag")
+    error: Optional[str] = Field(None, description="Error message if any")
+    start_time: Optional[datetime] = Field(None, description="Start time of the video")
+    end_time: Optional[datetime] = Field(None, description="End time of the video")
+    duration_sec: Optional[float] = Field(None, description="Duration of the video "
+                                                            "in seconds")
+
+
+# Define the data model for the QR record
+class QrRecord(BaseModel):
+    frame_start: int = Field(None, description="Frame number where QR code starts")
+    frame_end: int = Field(None, description="Frame number where QR code ends")
+    start_time: Optional[str] = Field(None, description="Time where QR code starts")
+    end_time: Optional[str] = Field(None, description="Time where QR code ends")
+    start_pos_sec: Optional[float] = Field(None, description="Position in seconds "
+                                                             "where QR code starts")
+    end_pos_sec: Optional[float] = Field(None, description="Position in seconds "
+                                                           "where QR code ends")
+    data: Optional[dict] = Field(None, description="QR code data")
+
+    def __str__(self):
+        return (f"QrRecord(frames=[{self.frame_start}, {self.frame_end}], "
+                f"pos=[{self.start_pos_sec}, {self.end_pos_sec} sec], "
+                f"start_time={self.start_time}, end_time={self.end_time}], "
```

Comment:
Let's uniformize
```suggestion
        return (f"QrRecord(frames=[{self.frame_start}, {self.frame_end}], "
                f"times=[{self.start_pos_sec}, {self.end_pos_sec} sec], "
                f"abs_times=[{self.start_time}, {self.end_time}], "
```

and make it a json single line record